chain/store/store.go

```diff
@@ -211,8 +213,8 @@ func (cs *ChainStore) SubHeadChanges(ctx context.Context) chan []*api.HeadChange
 	return out
 }
 
-func (cs *ChainStore) SubscribeHeadChanges(f func(rev, app []*types.TipSet) error) {
-	cs.headChangeNotifs = append(cs.headChangeNotifs, f)
+func (cs *ChainStore) SubscribeHeadChanges(f ReorgNotifee) {
+	cs.reorgNotifeeCh <- f
```

Comment:
Ideally I'd select from the send or a `closeCh`, but there are no lifecycle channels here I believe, and the `reorgWorker` uses a `context.TODO()` channel, which is just the process, so I don't expect this to hang.

Reply:
yeah, nbd not selecting on something here.